marketplace-and-integrations/packages/authorized-services.md

```diff
@@ -0,0 +1,776 @@
+
+## Features
+
+For the solution developer, the Umbraco Authorized Services offers two primary features.
+
+Firstly there's a tree available in the _Settings_ section of the backoffice, called _Authorized Services_. The tree shows the list of services based on the details provided in configuration.
+
+![authorized-services-tree](images/authorized-services-tree.png)
+
+Each tree entry has a management screen where an administrator can authenticate with an app that has been setup with the service. The status of each service is shown on this screen. When authorized, the authentication and authorization flow has been completed and an access token stored.
+
+![not-authorized-screen](images/not-authorized-screen.png)
+
+![authorized-screen](images/authorized-screen.png)
+
+Secondly, the developer has access to an interface - `IAuthorizedServiceCaller` - that they can inject instances of and use to make authorized requests to the service's API.
+
+Using a settings screen the administrator can review the service configuration.
+
+![settings-screen](images/settings-screen.png)
+
+## Usage
+
+### App Creation
+
+Services that this package are intended to support will offer an OAuth authentication and authorization flow against an "app". The developer will need to create this "app" with the service.  By doing this information such as the "client ID" and "client secret" can be applied in configuration.
+
+### Installation
+
+The package can be installed into your Umbraco solution from [NuGet](https://www.nuget.org/packages/Umbraco.AuthorizedServices).
+
+### Configuring a Service
+
+Details of services available need to be applied to the Umbraco web application's configuration, which, if using the `appSettings.json` file, will look as follows. Other sources such as environment variables can also be used, as per standard .NET configuration.
+
+```json
+  "Umbraco": {
+    "CMS": {
+        ...
+    },
+    "AuthorizedServices": {
+      "TokenEncryptionKey": "",
+      "Services": [
+        {
+          "Alias": "",
+          "DisplayName": "",
+          "ApiHost": "",
+          "IdentityHost": "",
+          "TokenHost": "",
+          "RequestIdentityPath": "",
+          "AuthorizationUrlRequiresRedirectUrl": true|false,
+          "RequestTokenPath": "",
+          "JsonSerializer": "",
+          "RequestTokenFormat": "",
+          "AuthorizationRequestRequiresAuthorizationHeaderWithBasicToken": true|false,
+          "ClientId": "",
+          "ClientSecret": "",
+          "UseProofKeyForCodeExchange": true|false,
+          "Scopes": "",
+          "AccessTokenResponseKey": "access_token",
+          "RefreshTokenResponseKey": "refresh_token",
+          "ExpiresInResponseKey": "expires_in",
+          "SampleRequest": ""
+        },
+      ]
+    }
+```
+
+#### Configuration Elements
```

Comment:
Comment for the following section:

Heading 5 (#####) as well as heading 6 (######) will not work on GitBook, as the max is 4.

Would it be an option to put the different config elements into a table instead?
It might also make it a little easier to read.
In a table, we could also add a column called "Required?" and then add yes/no accordingly 🤔 

Let me know what you think!

Reply:
Sounds good to me, I've reworked into a table.